src/clients/c++/request_common.cc

```diff
@@ -909,6 +895,12 @@ InferContextImpl::SetRunOptions(const InferContext::Options& boptions)
 
     auto routput = infer_request_.add_output();
     routput->set_name(output->Name());
+    if (ooptions.result_format == Result::ResultFormat::SHARED_MEMORY) {
```

Comment:
Use some other way to detect that you need to add these shared memory values.

Reply:
Will do